frontend/server/src/main/java/org/pytorch/serve/wlm/SequenceBatching.java

```diff
@@ -192,22 +202,38 @@ public void run() {
                     String jobGroupId =
                             eventJobGroupIds.poll(model.getMaxBatchDelay(), TimeUnit.MILLISECONDS);
                     if (jobGroupId == null || jobGroupId.isEmpty()) {
-                        CompletableFuture.runAsync(
-                                () -> {
-                                    try {
-                                        pollJobGroup();
-                                    } catch (InterruptedException e) {
-                                        logger.error("Failed to poll a job group", e);
-                                    }
-                                },
-                                pollExecutors);
+                        // Skip fetching new job groups when no capacity is available
+                        if (localCapacity.get() <= 0) {
+                            continue;
+                        }
+                        // Avoid duplicate poll tasks in the executor queue
+                        if (pollQueueTasks.containsKey("") && !pollQueueTasks.get("").isDone()) {
```

Comment:
How about naming the key pollJobGroup for doc purposes?

Reply:
Yes, that makes sense, will update the key.